### DISCUSSION TITLE
Is possible to auto load direnv entering in the project with it from neovim ?

### DISCUSSION THREAD

**Discussion by barcell1** [direnv/direnv#1343] [Q&A]
i know that is possible to go to terminal enter the project and then launch  neovim that would work.

But looking for a solution that im already in neovim in different project and go to a file from a project that contains direnv and it auto loads/activate

**Comment by ShreeChaturvedi** [comment]
The official [`direnv/direnv.vim`](https://github.com/direnv/direnv.vim) plugin does exactly this. In lazy.nvim:

```lua
{ "direnv/direnv.vim" }
```

Once installed, it hooks into `DirChanged` (neovim) and runs `direnv export vim` automatically whenever you open a file in a different project. So if you're editing something in project A and jump to a file in project B that has an `.envrc`, the environment updates on its own.

It's enabled by default (`g:direnv_auto = 1`). If you want it quieter, you can add `vim.g.direnv_silent_load = 1` to skip the echo output.
